contrib/initramfs/scripts/zfs

```diff
@@ -405,6 +405,8 @@ decrypt_fs()
 		ENCRYPTIONROOT="$(get_fs_value "${fs}" encryptionroot)"
 		KEYLOCATION="$(get_fs_value "${ENCRYPTIONROOT}" keylocation)"
 
+		echo -n "${ENCRYPTIONROOT}" > /run/zfs_fs_name
```

Comment:
I'm not sure that you ever answered whether the `-n` is really required on these. From my testing, they are not:

```
echo foo > /run/zfs_fs_name
zfs_fs_name=$(cat /run/zfs_fs_name)
echo x${zfs_fs_name}y
```

With `dash` and `bash`, I get "xfooy" with no space or newline.